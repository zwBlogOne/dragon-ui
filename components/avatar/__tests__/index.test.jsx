import React from 'react';
import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Avatar from '../index';

describe('Avatar', () => {
  it('renders Avatars of different size correctly', () => {
    const wrapper = render(
      <div>
        <Avatar size="default" />
        <Avatar size="64" />
        <Avatar size="large" />
        <Avatar size="small" />
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Avatars of different shape correctly', () => {
    const wrapper = shallow(
      <div>
        <Avatar size="circle" />
        <Avatar size="square" />
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Avatars of different type correctly', () => {
    const wrapper = render(
      <div>
        <Avatar icon="user" />
        <Avatar>U</Avatar>
        <Avatar>USER</Avatar>
        <Avatar src="http://img95.699pic.com/element/40044/5588.png_860.png" alt="my avatar" onError={() => console.log('load error')} />
        <Avatar style={{ color: '#2db7f5', backgroundColor: 'rgb(249, 232, 8)' }}>U</Avatar>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Avatars correctly when set src, icon and string', () => {
    const wrapper = render(
      <div>
        <Avatar src="http://img95.699pic.com/element/40044/5588.png_860.png" icon="user" >USER</Avatar>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
