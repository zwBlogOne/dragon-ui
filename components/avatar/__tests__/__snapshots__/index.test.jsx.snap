// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders Avatars correctly when set src, icon and string 1`] = `
<div>
  <span
    class="ui-avatar ui-avatar-circle ui-avatar-icon ui-avatar-image ui-avatar-string"
  >
    <img
      alt=""
      src="http://img95.699pic.com/element/40044/5588.png_860.png"
    />
  </span>
</div>
`;

exports[`Avatar renders Avatars of different shape correctly 1`] = `
<div>
  <Avatar
    alt=""
    icon=""
    onError={[Function]}
    prefixCls="ui-avatar"
    shape="circle"
    size="circle"
    src=""
  />
  <Avatar
    alt=""
    icon=""
    onError={[Function]}
    prefixCls="ui-avatar"
    shape="circle"
    size="square"
    src=""
  />
</div>
`;

exports[`Avatar renders Avatars of different size correctly 1`] = `
<div>
  <span
    class="ui-avatar ui-avatar-circle"
  />
  <span
    class="ui-avatar ui-avatar-circle"
  />
  <span
    class="ui-avatar ui-avatar-lg ui-avatar-circle"
  />
  <span
    class="ui-avatar ui-avatar-sm ui-avatar-circle"
  />
</div>
`;

exports[`Avatar renders Avatars of different type correctly 1`] = `
<div>
  <span
    class="ui-avatar ui-avatar-circle ui-avatar-icon"
  >
    <i
      class="ui-icon ui-icon-user theme-default"
    />
  </span>
  <span
    class="ui-avatar ui-avatar-circle ui-avatar-string"
  >
    <span
      style="position:absolute;transform:scale(0.9);left:calc(50% - 4.5px )"
    >
      U
    </span>
  </span>
  <span
    class="ui-avatar ui-avatar-circle ui-avatar-string"
  >
    <span
      style="position:absolute;transform:scale(0.6);left:calc(50% - 18px )"
    >
      USER
    </span>
  </span>
  <span
    class="ui-avatar ui-avatar-circle ui-avatar-image"
  >
    <img
      alt="my avatar"
      src="http://img95.699pic.com/element/40044/5588.png_860.png"
    />
  </span>
  <span
    class="ui-avatar ui-avatar-circle ui-avatar-string"
    style="color:#2db7f5;background-color:rgb(249, 232, 8)"
  >
    <span
      style="position:absolute;transform:scale(0.9);left:calc(50% - 4.5px )"
    >
      U
    </span>
  </span>
</div>
`;
